tests/framework/integration.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2022 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	"context"
+	"testing"
+
+	"go.etcd.io/etcd/client/pkg/v3/testutil"
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"go.etcd.io/etcd/tests/v3/framework/integration"
+	"go.etcd.io/etcd/tests/v3/framework/testutils"
+)
+
+type integrationFramework struct{}
+
+func (e integrationFramework) TestMain(m *testing.M) {
+	testutil.MustTestMainWithLeakDetection(m)
+}
+
+func (e integrationFramework) BeforeTest(t testing.TB) {
+	integration.BeforeTest(t)
+}
+
+func (e integrationFramework) NewCluster(t testing.TB) Cluster {
+	return &integrationCluster{
+		Cluster: integration.NewCluster(t, &integration.ClusterConfig{Size: 1}),
+		t:       t,
+	}
+}
+
+type integrationCluster struct {
+	*integration.Cluster
+	t testing.TB
+}
+
+func (c *integrationCluster) Close() error {
+	c.Terminate(c.t)
+	return nil
+}
+
+func (c *integrationCluster) Client() Client {
+	cc, err := c.ClusterClient()
+	if err != nil {
+		c.t.Fatal(err)
+	}
+	return &integrationClient{cc}
+}
+
+type integrationClient struct {
+	*clientv3.Client
+}
+
+func (c integrationClient) Get(key string, opts ...testutils.GetOption) (*clientv3.GetResponse, error) {
+	o := testutils.GetOptions{}
+	for _, opt := range opts {
+		opt(&o)
+	}
+	clientOpts := []clientv3.OpOption{}
+	if o.Serializable {
+		clientOpts = append(clientOpts, clientv3.WithSerializable())
+	}
+	return c.Client.Get(context.Background(), key, clientOpts...)
+}
+
+func (c integrationClient) Put(key, value string) error {
+	_, err := c.Client.Put(context.Background(), key, value)
```

Comment:
better not to ignore the first return value in framework, instead, let users to decide whether ignore it or not.

Reply:
The first argument is the PutResponse, I decided no not return it as most tests don't check the response assuming that request succeed. Implementing parsing PutResponse would make E2e client more complicated. Please remember that current interface is not final and is expect to evolve over time to handle new test cases.